app/domain/authentication/common/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,106 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    ExtractResourceRestrictions = CommandClass.new(
+      dependencies: {
+        resource_restrictions_class: ResourceRestrictions,
+        role_class:                  ::Role,
+        resource_class:              ::Resource,
+        logger:                      Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id host_name account)
+    ) do
+
+      def call
+        @logger.debug(LogMessages::Authentication::ExtractingRestrictionsFromResource.new(
+          @authenticator_name,
+          @host_name
+        ))
+
+        annotations = fetch_role_annotations
+        resource_restrictions_hash = extract_resource_restrictions_from_annotations(annotations)
+        resource_restrictions = create_resource_restrictions_object(resource_restrictions_hash)
+
+        @logger.debug(LogMessages::Authentication::ExtractedResourceRestrictions.new(resource_restrictions.names))
+
+        resource_restrictions
+      end
+
+      private
+
+      def fetch_role_annotations
+        # Validate role exists, otherwise getting role annotations return empty hash.
+        role_id = @role_class.roleid_from_username(@account, @host_name)
+        host_resource = @resource_class[role_id]
+
+        raise Errors::Authentication::Security::RoleNotFound, role_id unless host_resource
+
+        get_resource_annotations(host_resource)
+      end
+
+      def get_resource_annotations(host_resource)
```

Comment:
this should be `def resource_annotations(host_resource)`

Reply:
done